metricbeat/helper/prometheus/textparse_test.go

```diff
@@ -144,6 +144,166 @@ process_cpu 20
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(0)},
+				},
+				{
+					Name: stringp("nginx_sts_server_connects_total"),
+					Label: []*labels.Label{
+						{Name: "code", Value: "2xx"},
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(0)},
+				},
+				{
+					Name: stringp("nginx_sts_server_connects_total"),
+					Label: []*labels.Label{
+						{Name: "code", Value: "3xx"},
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(0)},
+				},
+				{
+					Name: stringp("nginx_sts_server_connects_total"),
+					Label: []*labels.Label{
+						{Name: "code", Value: "4xx"},
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(0)},
+				},
+				{
+					Name: stringp("nginx_sts_server_connects_total"),
+					Label: []*labels.Label{
+						{Name: "code", Value: "5xx"},
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(171)},
+				},
+				{
+					Name: stringp("nginx_sts_server_connects_total"),
+					Label: []*labels.Label{
+						{Name: "code", Value: "total"},
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(171)},
+				},
+			},
+		},
+		{
+			Name: stringp("nginx_sts_server_session_seconds_total"),
+			Help: stringp("The session duration time"),
+			Type: "counter",
+			Metric: []*OpenMetric{
+				{
+					Name: stringp("nginx_sts_server_session_seconds_total"),
+					Label: []*labels.Label{
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Counter: &Counter{Value: float64p(0.016)},
+				},
+			},
+		},
+		{
+			Name: stringp("nginx_sts_server_session_seconds"),
+			Help: stringp("The average of session duration time in seconds"),
+			Type: "gauge",
+			Metric: []*OpenMetric{
+				{
+					Name: stringp("nginx_sts_server_session_seconds"),
+					Label: []*labels.Label{
+						{Name: "listen", Value: "TCP:8091:127.0.0.1"},
+						{Name: "port", Value: "8091"},
+						{Name: "protocol", Value: "TCP"},
+					},
+					Gauge: &Gauge{Value: float64p(0.000)},
+				},
+			},
+		},
+	}
+
+	result, err := ParseMetricFamilies([]byte(input), ContentTypeTextFormat, time.Now(), nil)
+	if err != nil {
+		t.Fatalf("ParseMetricFamilies for content type %s returned an error.", ContentTypeTextFormat)
```

Comment:
[Blocker]
Please log the error as well. Otherwise if this test fail it'll be difficult to understand why it failed.

Reply:
@gpop63 could you please apply the requested change here?